### PR TITLE
API backend: Docker

### LINKED ISSUE #16: API backend: Docker
## Summary
Dockeurize the api, that can be running using flags.

Example of `api_server/docker/Dockerfile` :

```dockerfile
FROM ubuntu:latest

ARG DEBIAN_FRONTEND=noninteractive
ARG  PROJECT_NAME=purplecaffeine

LABEL org.opencontainers.image.title="${PROJECT_NAME}" \
      org.opencontainers.image.authors=${GITHUB_ACTOR} \
      org.opencontainers.image.vendor=${GITHUB_REPOSITORY} \
      org.opencontainers.image.source="${PROJECT_URL}" \
      org.opencontainers.image.url="${PROJECT_URL}/tags" \
      org.opencontainers.image.description="API purplecaffeine" \
      org.opencontainers.image.os="ubuntu" \
      org.opencontainers.image.version=${GITHUB_REF}

COPY . /opt/purplecaffeine

RUN apt-get update -yq \
&& apt-get install python3-pip -y

RUN pip3 install --upgrade pip setuptools
RUN pip3 install -r /opt/purplecaffeine/requirements.txt

WORKDIR /opt/purplecaffeine

VOLUME /etc/purplecaffeine/conf/
EXPOSE 8080

ENTRYPOINT [ "/opt/purplecaffeine" ]
CMD [ "--help" ]
```

And so that can be run by calling from `api_server/` :

```bash
docker run \
		--name purplecaffeine \
		-v $(PWD)/test:/etc/purplecaffeine/conf \
		-e PASS_DB=${PASS_DB} -e TOKEN_API=${TOKEN_API} \
		-p 8080:8080 \
		purplecaffeine:latest \
		-conf=/etc/purplecaffeine/conf/conf_docker.yaml \
		-swagger=/etc/purplecaffeine/conf/swagger.yaml
```

Or something similar.

### LINKED ISSUE #18: API backend: CI for docker
## Summary

Create the CI with GitHub to automate the build of the docker image and the push into our GitHub repo (or later DockerHub).

Example of CI :

```yaml
jobs:
  docker:
    name: Build purplecaffeine
    runs-on: ubuntu-latest
    permissions:
      packages: write
      contents: read
    env:
      IMAGE_NAME: "purplecaffeine_server"
      BASEFOLDER: "api_server"
    steps:
    - uses: actions/checkout@v2
    - name: Build image
      run: |
        cd ${{env.BASEFOLDER}}
        docker build . --file docker/Dockerfile --tag ${{ env.IMAGE_NAME }}:latest
    - name: Log in to registry
      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
    - name: Push image
      run: |
        IMAGE_ID=ghcr.io/${{ github.repository_owner }}/${{ env.IMAGE_NAME }}
        IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
        docker tag ${{ env.IMAGE_NAME }} $IMAGE_ID:latest
        docker push $IMAGE_ID:latest
```

Or something similar.

### PR DESCRIPTION
**Pull request by mickahell**
### Summary
Create the Docker image for production.
Add CI for building test.

### Details and comments
- [x] Create Dockerfile
- [x] Create admin user
- [x] Update README.md
- [x] Docker CI

---
Closes #16 
Closes #18

### CODE REVIEW

**Comment by mickahell** [comment]
@IceKhan13 ready for review :) I couldn't test the CI

File: .github/workflows/api_server-dockerhub-release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Release to DockerHub
+
+on:
+  workflow_dispatch:
+  push:
```
Comment (IceKhan13) [inline .github/workflows/api_server-dockerhub-release.yml]: Let's not push on tag for now and leave only workflow dispatch. Let's try to do couple of times manually and then we will switch to automatic pushes one we tested it manually first.

Let's use this action https://github.com/marketplace/actions/docker-build-push-action
It is much cleaner and doing everything for us.

**Review by IceKhan13 — CHANGES_REQUESTED** [review]
We can actually use your action too. Let's just remove on tag and we are good to go!

**Comment by mickahell** [comment]
@IceKhan13 ready 🚀 
I didn't knew this action, but it seems I'm doing exactly the same stuff. Even for tagging and version ^^
Anyway if it's ok to use like mine, I delete the push option :)

**Review by IceKhan13 — APPROVED** [review]
🚀

**Pull request merged**